src/core/common/message.cpp

```diff
@@ -133,7 +142,9 @@ void MessagePool::FreeBuffers(Buffer *aBuffer)
     while (aBuffer != nullptr)
     {
         Buffer *next = aBuffer->GetNextBuffer();
-#if OPENTHREAD_CONFIG_PLATFORM_MESSAGE_MANAGEMENT
+#if OPENTHREAD_CONFIG_MESSAGE_USE_HEAP_ENABLE
+        GetInstance().HeapFree(aBuffer);
+#elif OPENTHREAD_CONFIG_PLATFORM_MESSAGE_MANAGEMENT
```

Comment:
OPENTHREAD_CONFIG_PLATFORM_MESSAGE_MANAGEMENT should be of highest priority?

Reply:
updated to detect configuration errors.